app/src/main/java/com/gusta/wakemehome/DetailActivity.java

```diff
@@ -1,246 +1,356 @@
-        intent.setData(geoLocation);
-
-        if (intent.resolveActivity(getPackageManager()) != null) {
-            startActivity(intent);
-        } else {
-            Log.d(TAG, "Couldn't call " + geoLocation.toString() +
-                    ", no receiving apps installed!");
-        }
-    }
-
-    //=====================//
-    // OPTION MENU METHODS //
-    //=====================//
-
-    @Override
-    public boolean onCreateOptionsMenu(Menu menu) {
-        getMenuInflater().inflate(R.menu.menu_detail, menu);
-        return true;
-    }
-
-    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
-    @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
-        int id = item.getItemId();
-
-        if (id == R.id.action_save) {
-            onSaveButtonClicked();
-            return true;
-        }
-
-        return super.onOptionsItemSelected(item);
-    }
-}
+package com.gusta.wakemehome;
+
+import android.content.Context;
+import android.content.ContextWrapper;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.databinding.DataBindingUtil;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.gusta.wakemehome.database.AlarmEntry;
+import com.gusta.wakemehome.databinding.ActivityDetailBinding;
+import com.gusta.wakemehome.maps.MapAddress;
+import com.gusta.wakemehome.maps.MapsActivity;
+import com.gusta.wakemehome.viewmodel.AppExecutors;
+import com.gusta.wakemehome.viewmodel.DetailViewModel;
+
+import java.io.File;
+import java.util.Objects;
+
+public class DetailActivity extends AppCompatActivity {
+
+    //===========//
+    // CONSTANTS //
+    //===========//
+
+    // Constant for logging
+    private static final String TAG = DetailActivity.class.getSimpleName();
+    //default value for alarm id
+    public static final int DEFAULT_ALARM_ID = -1;
+    //Extra alarm id from main activity
+    public static final String EXTRA_ALARM_ID = "extraAlarmId";
+    //extra alarm from map provider
+    public static final String EXTRA_ALARM_COORDINATES = "alarmCoordinates";
+    //ui element which suppose to save on rotate
+    public static final String INSTANCE_ALARM_ID = "instanceAlarmId";
+    public static final String INSTANCE_ALARM_MESSAGE = "instanceAlarmMessage";
+    public static final String INSTANCE_ALARM_ALERT = "instanceAlarmAlert";
+    public static final String INSTANCE_ALARM_VIBRATE = "instanceAlarmVibrate";
+    public static final String INSTANCE_ALARM_ADDRESS_DATA = "instanceAlarmAddressData";

```

Comment:
Why do you need this? in my version they persist without the need to refer each and every one of them.